### ISSUE TITLE
t.words.test_service flaky with randomized hashes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5752]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5752|
|Type|defect|
|Created|2012-07-08 23:05:48Z|

The following test fails sporadically with ``python -R``:

```
[FAIL]
Traceback (most recent call last):
  File "/home/antoine/twisted/twisted/internet/defer.py", line 891, in _deferGenerator
    result = g.next()
  File "/home/antoine/twisted/twisted/words/test/test_service.py", line 419, in testJoin
    self.assertEqual(response[1][2], ['otheruser', '=', '#somechannel', 'firstuser otheruser'])
  File "/home/antoine/twisted/twisted/trial/unittest.py", line 270, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = ['otheruser', '=', '#somechannel', 'otheruser firstuser']
b = ['otheruser', '=', '#somechannel', 'firstuser otheruser']


twisted.words.test.test_service.IRCProtocolTestCase.testJoin
```

Attachments:

* [randomhash_test_service.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/192/1928265afceafded6ef7fadbc568ed9d69648c42/bc67dac84c211b720833f4e4ecb79be947830d09.patch) (700 bytes) - added by antoine on 2012-07-09 18:28:23Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5752 5752
type__defect defect
reporter__antoine antoine
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1341788748000000 1341788748000000
changetime__1342615303000000 1342615303000000
version__None None
owner__antoine antoine

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_2"></a>|@pitrou commented|
|-|-|

Here is a patch relaxing the ordering requirement in the test, and therefore fixing the failure.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_3"></a>|@ralphm set owner to @pitrou|
|-|-|

Looks good to me. Please add a news snippet and merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_4"></a>|@pitrou commented|
|-|-|

Replying to [ralphm](#note_3):
> Looks good to me. Please add a news snippet and merge. Thanks!

I can't merge since I don't have commit rights. As for the news snippet, I don't think it warrants one: it's just a test fix, not a bug fix.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_5"></a>|@ralphm set status to `closed`|
|-|-|

(In [34880]) Apply randomhash_test_service.patch: adjust test to consider nondeterministic order of IRC room occupants.

Author: antoine.
Reviewer: ralphm.
Fixes: [#5752](https://github.com/twisted/twisted/issues/5752).
